packages/kolme/src/core/kolme.rs

```diff
@@ -246,25 +246,26 @@ impl<App: KolmeApp> Kolme<App> {
     pub async fn propose_and_await_transaction(
         &self,
         tx: Arc<SignedTransaction<App::Message>>,
-    ) -> Result<Arc<SignedBlock<App::Message>>> {
+    ) -> std::result::Result<Arc<SignedBlock<App::Message>>, KolmeError> {
         let txhash = tx.hash();
         match tokio::time::timeout(
             self.tx_await_duration,
             self.propose_and_await_transaction_inner(tx),
         )
         .await
         {
-            Ok(res) => res,
-            Err(e) => Err(anyhow::Error::from(e).context(format!(
-                "Timed out proposing and awaiting transaction {txhash}"
-            ))),
+            Ok(res) => Ok(res?),
+            Err(e) => Err(KolmeError::TimeoutProposingTx {
+                txhash,
+                details: e.to_string(),
```

Comment:
If possible, keeping the original error type would be better than serializing to String.